packages/nx-plugin/src/executors/autorun/executor.ts

```diff
@@ -0,0 +1,44 @@
+import { logger } from '@nx/devkit';
+// eslint-disable-next-line n/no-sync
+import { execSync } from 'node:child_process';
+import { ExecutorContext } from 'nx/src/config/misc-interfaces';
+import { createCliCommand } from '../internal/cli';
+import { normalizeContext } from '../internal/context';
+import { AUTORUN_COMMAND } from './constants';
+import { AutorunCommandExecutorOptions } from './schema';
+import { parseAutorunExecutorOptions } from './utils';
+
+export default runAutorunExecutor;
+
+export function runAutorunExecutor(
+  options: AutorunCommandExecutorOptions,
+  context: ExecutorContext,
+) {
+  const normalizedContext = normalizeContext(context);
+  const cliArgumentObject = parseAutorunExecutorOptions(
+    options,
+    normalizedContext,
+  );
+  const command = createCliCommand(AUTORUN_COMMAND, cliArgumentObject);
+
+  const { dryRun } = options;
+  if (dryRun) {
+    logger.warn(`DryRun execution of: ${command}`);
+  } else {
+    try {
+      // eslint-disable-next-line n/no-sync
+      execSync(command, context.cwd ? { cwd: context.cwd } : {});
```

Comment:
Why not just use the async version? Is there some reason this needs to block the main thread?

Reply:
I wanted to keep that out of this PR as it includes proper forwarding of the logs and errors. Mentioned it in the PR description now.